applications/GeoMechanicsApplication/custom_strategies/schemes/generalized_newmark_scheme.hpp

```diff
@@ -21,58 +21,100 @@ class GeneralizedNewmarkScheme
 public:
-    GeneralizedNewmarkScheme(double theta,
-                             const Variable<double>& rVariable,
-                             const Variable<double>& rDeltaTimeVariable,
-                             const Variable<double>& rDeltaTimeVariableCoefficient)
-        : GeoMechanicsTimeIntegrationScheme<TSparseSpace, TDenseSpace>(),
+    GeneralizedNewmarkScheme(const std::vector<FirstOrderScalarVariable>& rFirstOrderScalarVariables,
+                             const std::vector<SecondOrderVectorVariable>& rSecondOrderVectorVariables,
+                             double theta,
+                             double beta,
+                             double gamma)
+        : GeoMechanicsTimeIntegrationScheme<TSparseSpace, TDenseSpace>(
+              rFirstOrderScalarVariables, rSecondOrderVectorVariables),
           mTheta(theta),
-          mVariable(rVariable),
-          mDeltaTimeVariable(rDeltaTimeVariable),
-          mDeltaTimeVariableCoefficient(rDeltaTimeVariableCoefficient)
+          mBeta(beta),
+          mGamma(gamma)
     {
-        KRATOS_ERROR_IF(this->mTheta <= 0)
-            << "Theta must be larger than zero, but got " << this->mTheta << "\n";
+        KRATOS_ERROR_IF(mTheta <= 0)
+            << "Theta must be larger than zero, but got " << mTheta << "\n";
+        KRATOS_ERROR_IF(mBeta <= 0)
+            << "Beta must be larger than zero, but got " << mBeta << "\n";
+        KRATOS_ERROR_IF(mGamma <= 0)
+            << "Gamma must be larger than zero, but got " << mGamma << "\n";
     }
 
-protected:
-    void CheckAllocatedVariables(const ModelPart& rModelPart) const override
+    GeneralizedNewmarkScheme(const std::vector<FirstOrderScalarVariable>& rFirstOrderScalarVariables,
+                             double theta)
+        : GeoMechanicsTimeIntegrationScheme<TSparseSpace, TDenseSpace>(
+              rFirstOrderScalarVariables, {}),
+          mTheta(theta)
     {
-        for (const auto& r_node : rModelPart.Nodes())
-        {
-            this->CheckSolutionStepsData(r_node, mVariable);
-            this->CheckSolutionStepsData(r_node, mDeltaTimeVariable);
-            this->CheckDof(r_node, mVariable);
-        }
+        KRATOS_ERROR_IF(mTheta <= 0)
+            << "Theta must be larger than zero, but got " << mTheta << "\n";
     }
 
-    inline void SetTimeFactors(ModelPart& rModelPart) override
+    GeneralizedNewmarkScheme(const std::vector<SecondOrderVectorVariable>& rSecondOrderVectorVariables,
+                             double beta,
+                             double gamma)
+        : GeoMechanicsTimeIntegrationScheme<TSparseSpace, TDenseSpace>(
+              {}, rSecondOrderVectorVariables),
+          mBeta(beta),
+          mGamma(gamma)
+    {
+        KRATOS_ERROR_IF(mBeta <= 0)
+            << "Beta must be larger than zero, but got " << mBeta << "\n";
+        KRATOS_ERROR_IF(mGamma <= 0)
+            << "Gamma must be larger than zero, but got " << mGamma << "\n";
+    }
```

Comment:
Since this constructor is not used (and probably also not tested), I feel we'd better remove this one. If we need it later, we can add it then.

Reply:
Agreed, removed